numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1769,6 +1770,79 @@ array_free(PyObject * v)
     PyObject_Free(v);
 }
 
+static const char *
+_attr_name_as_utf8(PyObject *attr_name)
+{
+    const char *c_attr_name;
+    if (PyUString_Check(attr_name)) {
+        c_attr_name = PyUString_AsUTF8(attr_name);
```

Comment:
actually, on second thought, since in the end we care about python3 more than python2, can one just do this in "try/except" form:
```
    const char *c_attr_name = PyUString_AsUTF8(attr_name);
#if !defined(NPY_PY3K)
    if (c_attr_name == NULL)
    {
        PyErr_Clear();
        c_attr_name = PyString_AsString(attr_name);
    }
#endif
    return c_attr_name;
}
```
(I think in this case no exception needs to be set any more)